api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -953,8 +953,13 @@ message Cluster {
   // :ref:`STRICT_DNS<envoy_v3_api_enum_value_config.cluster.v3.Cluster.DiscoveryType.STRICT_DNS>`
   // and :ref:`LOGICAL_DNS<envoy_v3_api_enum_value_config.cluster.v3.Cluster.DiscoveryType.LOGICAL_DNS>`
   // this setting is ignored.
-  google.protobuf.Duration dns_refresh_rate = 16
-      [(validate.rules).duration = {gt {nanos: 1000000}}];
+  // This field is deprecated in favor of :ref:`cluster_type<envoy_v3_api_field_config.cluster.v3.Cluster.cluster_type>` and
```

Comment:
I think this same comment should apply to all DNS-related fields in the `Cluster` proto (e.g., `dns_lookup_family`, `dns_resolvers`, etc).

We'll need to add corresponding fields to the new `DnsCluster` message for each of those fields.

Reply:
Yes agreed.

Reply:
@markdroth @mattklein redis cluster uses these fields, so we will have to update that too?

https://github.com/envoyproxy/envoy/blob/b28b7f6d8945d978324a13213e89fedcc45470ac/source/extensions/clusters/redis/redis_cluster.cc#L61-L62

Reply:
I'm not that familiar with the redis cluster type, but I suspect that we do want it to have its own parameters for DNS configuration.  But I would also be okay with deferring some of that to a separate PR: we could document that these fields are deprecated for the STRICT_DNS and LOGICAL_DNS cluster types only for now, but add TODO comments indicating that we will eventually deprecate them for redis as well.